S_RTOS_01IntroductionToFreertos/Task_3/button_task/button_task.c

```diff
@@ -0,0 +1,35 @@
+#include "button_task.h"
+
+
+void button_task(void *pvParameters){
+
+	
+	pinState_t button_state;
+	uint8_t sec_counter =0;
+	
+	for(;;){
+		button_state = GPIO_read(PORT_0, ((button_task_config *)pvParameters)->pin_num);
+		if(button_state == PIN_IS_HIGH){
+			
+			while(button_state == PIN_IS_HIGH){
+				vTaskDelay(1000);
+				sec_counter++;
+				button_state = GPIO_read(PORT_0, ((button_task_config *)pvParameters)->pin_num);
+			}
+			if(sec_counter>4){
+				*(((button_task_config *)pvParameters)->led_delay)= 100;
+			}else if((sec_counter>2) && (sec_counter<4)){
```

Comment:
don't use magic number

Reply:
ok, thanks for your interest, I will resolve it.